src/System.Linq/samples/project.json

```diff
@@ -0,0 +1,13 @@
+{
+  "dependencies": {
+    "System.Collections": "4.0.10",
```

Comment:
This is sort of what I was alluding to when I brought up project-vs-nuget references in the other issue. Even a "simple" project like this one will bring in a dozen+ dependencies which are grabbed via NuGet. I can imagine some scenarios where we are transitively depending on a few unreleased contracts, which could cause some problems, depending on how the project references are set up. If the pre-release library is pulled in through a nuget dependency, then we might not catch an incompatibility until after new packages are published, or until after dependencies are re-locked (assuming we are using * versioning with a locked project.lock.json). If the library is pulled in through a project reference, then people doing completely unrelated work would get snagged by a random samples project failing to build. That may be what we want, I just want to make sure that this has been discussed, as it's something that will affect developer workflow.